tf/tfprocess.py

```diff
@@ -135,10 +135,15 @@ def init_net(self, next_batch):
         self.batch_norm_count = 0
         self.y_conv, self.z_conv = self.construct_net(self.x)
 
-        # y_ has -1 on illegal moves, flush them to 0 first
-        self.y_ = tf.nn.relu(self.y_)
-
         # Calculate loss on policy head
+        if self.cfg['training'].get('mask_legal_moves'):
+            # extract mask for legal moves from target policy
+            move_is_legal = tf.greater_equal(self.y_, 0)
```

Comment:
How confident are we that this works correctly with both y_ as negative and y_ as NaN?

Reply:
Quite confident: https://colab.research.google.com/drive/1jCLSgdkITC3rVN4qau6AQAZujAOvBsWo NaNs tend to compare false to everything, so this should be standard behaviour.
I don't know why our cross-entropy isn't NaN, it might be worth adding wrapping `self.y_` in a `tf.where(tf.is_nan(self.y_), tf.zeros_like(self.y_), self.y_)`

Reply:
I'm more confident that I'd notice if the cross entropy started calculating NaN in the presence of NaN training targets - so I think we can live with the current status there while we fix lc0 to not output NaN.